app/uk/gov/hmrc/helptosavetestadminfrontend/util/TotpGenerator.scala

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright 2018 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.helptosavetestadminfrontend.util
+
+import java.security.Key
+import javax.crypto.Mac
+import javax.crypto.spec.SecretKeySpec
+
+import org.apache.commons.codec.binary.Base32
+
+import scala.concurrent.duration._
+import scala.math._
+
+sealed trait CryptoAlgorithm
+case object HmacSHA512 extends CryptoAlgorithm
+
+trait TotpGenerator {
```

Comment:
could we not bring in the TOTP generator stuff as a dependency through `build.sbt`?

Reply:
yeah I've now bought the TOTP generator stuff in via dependency and removed the trait I created